### DISCUSSION TITLE
How to restrict the list of users that apear in the 'owners' filter for dashboards/charts?

### DISCUSSION THREAD

**Discussion by gervarela** [apache/superset#19757] [Q&A / Help]
By default, even a Gamma (or more restricted user) has access to the complete list of users throught the 'Owner' and 'Created By' filters in the Dashboards and Charts main pages.

Is there a way to restrict this list to users with the same roles? o Even disable these filters completely?

**Comment by nytai** [comment]
You can always just edit the code and remove the filters. All of these list view pages are here https://github.com/apache/superset/tree/master/superset-frontend/src/views/CRUD

**Reply by gervarela** [reply]
Thanks for the answer, that will be an option, but probably the last one... if there is no other chance.

What don't make many sense to me is that a user with restrictec access to datasets/dashboards, gets the full list of users. I think that this behaviour has many privacy and usability issues associated.

In fact, looking at the code of the backend, I don't know if this is really the expected behaviour, because it seems to try to obtaing the 'related' owners of the Dashboards, but in reality, it is getting all the users, even users that do not own any dashboard at all.

I whink that a more 'logical' approach would be to return only the users related (as owners) to the dashboards (graphs, etc.) that are being shown, or the users has access to.

Don't know if this qualifies as an issue or a feature request.

**Reply by shubhamofbce** [reply]
@gervarela Do you have any solutions for this apart from editing the code ??
